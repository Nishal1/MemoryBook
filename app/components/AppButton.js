import { StyleSheet, 
    Text, 
    View, 
    TouchableOpacity} from 'react-native';
import React from 'react';

import ColorPicker from '../config/ColorPicker';

export default function AppButton(props) {
  return (
    <TouchableOpacity onPress={props.onPress}>
        <View style={props.color? [styles.button, {backgroundColor: ColorPicker[props.color]}] :styles.button}>
            <Text style={styles.text}>{props.title}</Text>
        </View>
    </TouchableOpacity>
  )
}

const styles = StyleSheet.create({
    button: {
        backgroundColor: ColorPicker.secondaryColor,
        borderRadius: 15,
        width: '100%',
        padding: 15,
        justifyContent: 'center',
        alignItems: 'center',
        marginTop: 5
    },
    text: {
        color: ColorPicker.white,
        fontSize: 14,
        textTransform: 'uppercase',
        fontWeight: 'bold'
    }
})