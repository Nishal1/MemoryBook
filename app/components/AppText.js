import { StyleSheet, Text, View } from 'react-native'
import React from 'react'

export default function AppText(props) {
  return (
      <Text style={props.style ? [styles.text, props.style]: styles.text}>
          {props.children}
      </Text>
  )
}

const styles = StyleSheet.create({
    text: {
        fontSize: 20,
        fontFamily: 'Roboto'
    }
})