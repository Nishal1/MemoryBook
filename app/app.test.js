import React from 'react';
import renderer from 'react-test-renderer';
import AppText from './components/AppText';
import AppButton from './components/AppButton';
import StyleText from './components/StyleText';
import ValidationMessage from './components/ValidationMessage';
import ColorPicker from './config/ColorPicker';

import LoginScreen from './screens/LoginScreen';
import RegisterScreen from './screens/RegisterScreen';
import WelcomeScreen from './screens/WelcomeScreen';

import { authenticate, isUniqueUser, deleteMemory,
    beginSession, addMemory, getImgs,
    getCurrUser, endSession, registerUser } from './controller/logic';

//logic tests
it('#LG:01 Logging in and logging out test', () => {
    //a sample user that aldready exists
    const user = {
        username: "ross",
        password: "09876"
    };
    beginSession(user);
    expect(authenticate(user)).toBe(true);
    expect(getCurrUser().username).toBe("ross");
    endSession();
});

it('#LG:02 Registration test', () => {
    //new user
    const newUser = {
        email: 'newuser@gmail.com',
        username: 'newuser',
        name: 'New User',
        password: 'xxxx'
    };
    //already existing user
    const oldUser = {
        username: "ross",
        password: "09876"
    };

    expect(isUniqueUser(newUser)).toBe(true); //new user can register into the system
    expect(isUniqueUser(oldUser)).toBe(false); //old user cannot register again

    registerUser(newUser);
    expect(getCurrUser().username).toBe(newUser.username); //make sure new user is logged in 
    //after registering
});

it('#LG:03 Add memory test', () => {
    const user = {
        username: "ross",
        password: "09876"
    };
    beginSession(user);
    expect(authenticate(user)).toBe(true);
    //new memory to add
    const newMemory = {
        id: "nfosh211212",
        title: 'new memory',
        image: require('./assets/defaultProfile.png'),
        category: 'Fun',
        password: 'xxxx'
    };
    addMemory(newMemory, getCurrUser().id);
    let newMem = getImgs().find((i) => i.id === "nfosh211212");
    expect(newMem.id).toBe("nfosh211212"); //make sure new memory exists
    expect(newMem.userid).toBe(getCurrUser().id);

    //log the user out and sign back in to test if new memory still persists
    endSession();
    expect(getCurrUser()).toBe(null);
    beginSession(user);
    expect(authenticate(user)).toBe(true);
    expect(getCurrUser().username).toBe("ross");

    newMem = getImgs().find((i) => i.id === "nfosh211212")
    expect(newMem.id).toBe("nfosh211212"); //make sure new memory exists
});

//UI tests
test("#UI.01 AppText has correct font family and size", () => {
    const json = renderer.create(<AppText />).toJSON();
    expect(json.props.style.fontSize).toBe(20);
    expect(json.props.style.fontFamily).toBe('Roboto');
});

test("#UI.02 AppText renders text correctly", () => {
    const json = renderer.create(<AppText>Some text is here!</AppText>).toJSON();
    expect(json.props.style.fontSize).toBe(20);
    expect(json.props.style.fontFamily).toBe('Roboto');
    expect(json.children.includes("Some text is here!"));
});

test("#UI.03 StyleText renders text correctly", () => {
    const json = renderer.create(<StyleText>Welcome to memorybook!</StyleText>).toJSON();
    expect(json.children.includes("Welcome to memorybook!"));
    expect(json.children[0].props.style.fontFamily).toBe('Roboto');
    expect(json.children[0].props.style.fontSize).toBe(25);
});

test("#UI.04 Register button renders correctly", () => {
    const json = renderer.create(
        <AppButton 
            title="Register"
            onPress={() => navigation.navigate("Register")}
            color="otherColor2"
        />).toJSON();
    expect(json.children[0].children.includes('Register'));
    expect(json.children[0].props.style[1].backgroundColor).toBe('#8C8C9C');
    expect(json.children[0].props.style[0].borderRadius).toBe(15);
});

test("#UI.05 Login button renders correctly", () => {
    const json = renderer.create(
        <AppButton 
            title="Login"
            onPress={() => navigation.navigate("Login")}
        />).toJSON();
    expect(json.children[0].children.includes('Login'));
    expect(json.children[0].props.style.backgroundColor).toBe('#BBB6A5');
    expect(json.children[0].props.style.borderRadius).toBe(15);
});

test("#UI.06 Validation error message renders text correctly", () => {
    const json = renderer.create(<ValidationMessage text="Error message goes here" />).toJSON();
    expect(json.children.includes("Error message goes here"));
    expect(json.props.style.backgroundColor).toBe('#F7B0BB');
    expect(json.props.style.borderColor).toBe("#C82333");
});

test("#UI.07 The color scheme is correct", () => {
    const {primaryColor, secondaryColor,
        otherColor1, otherColor2, 
        inActiveColor, offWhite, khaki,
        red} = ColorPicker;
    expect(primaryColor).toBe('#2A2F33');
    expect(secondaryColor).toBe('#BBB6A5');
    expect(otherColor1).toBe('#8C9491');
    expect(otherColor2).toBe('#8C8C9C');
    expect(inActiveColor).toBe('#37414a');
    expect(offWhite).toBe('#f5f5f5');
    expect(khaki).toBe('#B5A197');
    expect(red).toBe('#C82333');
});

//snapshot tests
it('#SN.01 Login screen renders correctly', () => {
    const tree = renderer
      .create(<LoginScreen />)
      .toJSON();
    expect(tree).toMatchSnapshot();
});

it('#SN.02 Register screen renders correctly', () => {
    const tree = renderer
      .create(<RegisterScreen />)
      .toJSON();
    expect(tree).toMatchSnapshot();
});

it('#SN.03 Welcome screen renders correctly', () => {
    const tree = renderer
      .create(<WelcomeScreen />)
      .toJSON();
    expect(tree).toMatchSnapshot();
});


