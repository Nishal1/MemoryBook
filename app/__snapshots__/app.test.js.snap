// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#SN.01 Login screen renders correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Array [
      Object {
        "flex": 1,
        "marginTop": 20,
      },
      Object {
        "flex": 1,
        "marginTop": 0,
      },
    ]
  }
>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      accessibilityIgnoresInvertColors={true}
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <Image
        blurRadius={0.9}
        source={1}
        style={
          Array [
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
            Object {
              "height": undefined,
              "width": undefined,
            },
            undefined,
          ]
        }
      />
      <RCTScrollView>
        <View>
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#f5f5f5",
                "borderColor": "#B5A197",
                "borderWidth": 10,
                "elevation": 20,
                "height": 550,
                "justifyContent": "center",
                "margin": 20,
                "marginTop": "25%",
                "padding": 10,
                "shadowColor": "#000",
              }
            }
          >
            <View>
              <Text
                style={
                  Object {
                    "fontFamily": "Roboto",
                    "fontSize": 25,
                  }
                }
              >
                Login
              </Text>
            </View>
            <View
              style={
                Object {
                  "marginBottom": 30,
                  "marginTop": 50,
                }
              }
            >
              <View
                style={
                  Object {
                    "backgroundColor": "#e0e0eb",
                    "borderRadius": 25,
                    "flexDirection": "row",
                    "marginVertical": 10,
                    "padding": 10,
                    "width": "100%",
                  }
                }
              >
                <Text />
                <TextInput
                  allowFontScaling={true}
                  autoCapitalize="none"
                  autoCorrect={false}
                  keyboardType="default"
                  onBlur={[Function]}
                  onChangeText={[Function]}
                  placeholder="Username"
                  rejectResponderTermination={true}
                  style={
                    Object {
                      "flex": 1,
                      "fontFamily": "Roboto",
                      "fontSize": 20,
                      "marginLeft": 10,
                    }
                  }
                  underlineColorAndroid="transparent"
                  value=""
                />
              </View>
              <View
                style={
                  Object {
                    "backgroundColor": "#e0e0eb",
                    "borderRadius": 25,
                    "flexDirection": "row",
                    "marginVertical": 10,
                    "padding": 10,
                    "width": "100%",
                  }
                }
              >
                <Text />
                <TextInput
                  allowFontScaling={true}
                  autoCapitalize="none"
                  autoCorrect={false}
                  keyboardType="default"
                  onBlur={[Function]}
                  onChangeText={[Function]}
                  placeholder="Password"
                  rejectResponderTermination={true}
                  secureTextEntry={true}
                  style={
                    Object {
                      "flex": 1,
                      "fontFamily": "Roboto",
                      "fontSize": 20,
                      "marginLeft": 10,
                    }
                  }
                  underlineColorAndroid="transparent"
                  value=""
                />
              </View>
            </View>
            <View>
              <View
                accessible={true}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "opacity": 1,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "#BBB6A5",
                      "borderRadius": 15,
                      "flexDirection": "row",
                      "justifyContent": "center",
                      "marginTop": 5,
                      "padding": 15,
                      "width": "100%",
                    }
                  }
                >
                  <Text
                    style={
                      Object {
                        "color": "#fff",
                        "fontSize": 14,
                        "fontWeight": "bold",
                        "textTransform": "uppercase",
                      }
                    }
                  >
                    Login
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </View>
      </RCTScrollView>
    </View>
  </View>
</RCTSafeAreaView>
`;

exports[`#SN.02 Register screen renders correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Array [
      Object {
        "flex": 1,
        "marginTop": 20,
      },
      Object {
        "flex": 1,
        "marginTop": 0,
      },
    ]
  }
>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      accessibilityIgnoresInvertColors={true}
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <Image
        blurRadius={0.9}
        source={1}
        style={
          Array [
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
            Object {
              "height": undefined,
              "width": undefined,
            },
            undefined,
          ]
        }
      />
      <RCTScrollView>
        <View>
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#f5f5f5",
                "borderColor": "#B5A197",
                "borderWidth": 10,
                "elevation": 20,
                "height": 700,
                "justifyContent": "center",
                "margin": 20,
                "padding": 10,
                "shadowColor": "#000",
              }
            }
          >
            <View>
              <Text
                style={
                  Object {
                    "fontFamily": "Roboto",
                    "fontSize": 25,
                  }
                }
              >
                Register
              </Text>
            </View>
            <View
              style={
                Object {
                  "marginBottom": 30,
                  "marginTop": 50,
                }
              }
            >
              <View
                style={
                  Object {
                    "backgroundColor": "#e0e0eb",
                    "borderRadius": 25,
                    "flexDirection": "row",
                    "marginVertical": 10,
                    "padding": 10,
                    "width": "100%",
                  }
                }
              >
                <Text />
                <TextInput
                  allowFontScaling={true}
                  autoCapitalize="none"
                  autoCorrect={false}
                  keyboardType="email-address"
                  onBlur={[Function]}
                  onChangeText={[Function]}
                  placeholder="Email Address"
                  rejectResponderTermination={true}
                  style={
                    Object {
                      "flex": 1,
                      "fontFamily": "Roboto",
                      "fontSize": 20,
                      "marginLeft": 10,
                    }
                  }
                  underlineColorAndroid="transparent"
                  value=""
                />
              </View>
              <View
                style={
                  Object {
                    "backgroundColor": "#e0e0eb",
                    "borderRadius": 25,
                    "flexDirection": "row",
                    "marginVertical": 10,
                    "padding": 10,
                    "width": "100%",
                  }
                }
              >
                <Text />
                <TextInput
                  allowFontScaling={true}
                  autoCapitalize="none"
                  autoCorrect={false}
                  keyboardType="default"
                  onBlur={[Function]}
                  onChangeText={[Function]}
                  placeholder="Username"
                  rejectResponderTermination={true}
                  style={
                    Object {
                      "flex": 1,
                      "fontFamily": "Roboto",
                      "fontSize": 20,
                      "marginLeft": 10,
                    }
                  }
                  underlineColorAndroid="transparent"
                  value=""
                />
              </View>
              <View
                style={
                  Object {
                    "backgroundColor": "#e0e0eb",
                    "borderRadius": 25,
                    "flexDirection": "row",
                    "marginVertical": 10,
                    "padding": 10,
                    "width": "100%",
                  }
                }
              >
                <Text />
                <TextInput
                  allowFontScaling={true}
                  autoCapitalize="words"
                  autoCorrect={false}
                  keyboardType="default"
                  onBlur={[Function]}
                  onChangeText={[Function]}
                  placeholder="Full Name"
                  rejectResponderTermination={true}
                  style={
                    Object {
                      "flex": 1,
                      "fontFamily": "Roboto",
                      "fontSize": 20,
                      "marginLeft": 10,
                    }
                  }
                  underlineColorAndroid="transparent"
                  value=""
                />
              </View>
              <View
                style={
                  Object {
                    "backgroundColor": "#e0e0eb",
                    "borderRadius": 25,
                    "flexDirection": "row",
                    "marginVertical": 10,
                    "padding": 10,
                    "width": "100%",
                  }
                }
              >
                <Text />
                <TextInput
                  allowFontScaling={true}
                  autoCapitalize="none"
                  autoCorrect={false}
                  keyboardType="default"
                  onBlur={[Function]}
                  onChangeText={[Function]}
                  placeholder="Password"
                  rejectResponderTermination={true}
                  secureTextEntry={true}
                  style={
                    Object {
                      "flex": 1,
                      "fontFamily": "Roboto",
                      "fontSize": 20,
                      "marginLeft": 10,
                    }
                  }
                  underlineColorAndroid="transparent"
                  value=""
                />
              </View>
            </View>
            <View
              style={
                Object {
                  "marginBottom": 10,
                }
              }
            >
              <View
                accessible={true}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "opacity": 1,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "#BBB6A5",
                      "borderRadius": 15,
                      "flexDirection": "row",
                      "justifyContent": "center",
                      "marginTop": 5,
                      "padding": 15,
                      "width": "100%",
                    }
                  }
                >
                  <Text
                    style={
                      Object {
                        "color": "#fff",
                        "fontSize": 14,
                        "fontWeight": "bold",
                        "textTransform": "uppercase",
                      }
                    }
                  >
                    Register
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </View>
      </RCTScrollView>
    </View>
  </View>
</RCTSafeAreaView>
`;

exports[`#SN.03 Welcome screen renders correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Array [
      Object {
        "flex": 1,
        "marginTop": 20,
      },
      undefined,
    ]
  }
>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      accessibilityIgnoresInvertColors={true}
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <Image
        blurRadius={0.9}
        source={1}
        style={
          Array [
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
            Object {
              "height": undefined,
              "width": undefined,
            },
            undefined,
          ]
        }
      />
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#f5f5f5",
            "borderColor": "#B5A197",
            "borderWidth": 10,
            "elevation": 20,
            "flex": 0.7,
            "justifyContent": "center",
            "margin": 20,
            "marginTop": "40%",
            "shadowColor": "#000",
          }
        }
      >
        <Text />
        <View
          style={
            Object {
              "fontFamily": "Roboto",
            }
          }
        >
          <View>
            <Text
              style={
                Object {
                  "fontFamily": "Roboto",
                  "fontSize": 25,
                }
              }
            >
              Welcome to memorybook!
            </Text>
          </View>
          <Text
            style={
              Array [
                Object {
                  "fontFamily": "Roboto",
                  "fontSize": 20,
                },
                Object {
                  "alignItems": "center",
                  "fontStyle": "italic",
                  "justifyContent": "center",
                },
              ]
            }
          >
            "One Solution to storing photos"
          </Text>
        </View>
        <View
          style={
            Object {
              "margin": 10,
            }
          }
        >
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#BBB6A5",
                  "borderRadius": 15,
                  "flexDirection": "row",
                  "justifyContent": "center",
                  "marginTop": 5,
                  "padding": 15,
                  "width": "100%",
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "#fff",
                    "fontSize": 14,
                    "fontWeight": "bold",
                    "textTransform": "uppercase",
                  }
                }
              >
                Login
              </Text>
            </View>
          </View>
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "#BBB6A5",
                    "borderRadius": 15,
                    "flexDirection": "row",
                    "justifyContent": "center",
                    "marginTop": 5,
                    "padding": 15,
                    "width": "100%",
                  },
                  Object {
                    "backgroundColor": "#8C8C9C",
                  },
                ]
              }
            >
              <Text
                style={
                  Object {
                    "color": "#fff",
                    "fontSize": 14,
                    "fontWeight": "bold",
                    "textTransform": "uppercase",
                  }
                }
              >
                Register
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</RCTSafeAreaView>
`;

exports[`Login screen renders correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Array [
      Object {
        "flex": 1,
        "marginTop": 20,
      },
      Object {
        "flex": 1,
        "marginTop": 0,
      },
    ]
  }
>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      accessibilityIgnoresInvertColors={true}
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <Image
        blurRadius={0.9}
        source={1}
        style={
          Array [
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
            Object {
              "height": undefined,
              "width": undefined,
            },
            undefined,
          ]
        }
      />
      <RCTScrollView>
        <View>
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#f5f5f5",
                "borderColor": "#B5A197",
                "borderWidth": 10,
                "elevation": 20,
                "height": 550,
                "justifyContent": "center",
                "margin": 20,
                "marginTop": "25%",
                "padding": 10,
                "shadowColor": "#000",
              }
            }
          >
            <View>
              <Text
                style={
                  Object {
                    "fontFamily": "Roboto",
                    "fontSize": 25,
                  }
                }
              >
                Login
              </Text>
            </View>
            <View
              style={
                Object {
                  "marginBottom": 30,
                  "marginTop": 50,
                }
              }
            >
              <View
                style={
                  Object {
                    "backgroundColor": "#e0e0eb",
                    "borderRadius": 25,
                    "flexDirection": "row",
                    "marginVertical": 10,
                    "padding": 10,
                    "width": "100%",
                  }
                }
              >
                <Text />
                <TextInput
                  allowFontScaling={true}
                  autoCapitalize="none"
                  autoCorrect={false}
                  keyboardType="default"
                  onBlur={[Function]}
                  onChangeText={[Function]}
                  placeholder="Username"
                  rejectResponderTermination={true}
                  style={
                    Object {
                      "flex": 1,
                      "fontFamily": "Roboto",
                      "fontSize": 20,
                      "marginLeft": 10,
                    }
                  }
                  underlineColorAndroid="transparent"
                  value=""
                />
              </View>
              <View
                style={
                  Object {
                    "backgroundColor": "#e0e0eb",
                    "borderRadius": 25,
                    "flexDirection": "row",
                    "marginVertical": 10,
                    "padding": 10,
                    "width": "100%",
                  }
                }
              >
                <Text />
                <TextInput
                  allowFontScaling={true}
                  autoCapitalize="none"
                  autoCorrect={false}
                  keyboardType="default"
                  onBlur={[Function]}
                  onChangeText={[Function]}
                  placeholder="Password"
                  rejectResponderTermination={true}
                  secureTextEntry={true}
                  style={
                    Object {
                      "flex": 1,
                      "fontFamily": "Roboto",
                      "fontSize": 20,
                      "marginLeft": 10,
                    }
                  }
                  underlineColorAndroid="transparent"
                  value=""
                />
              </View>
            </View>
            <View>
              <View
                accessible={true}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "opacity": 1,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "#BBB6A5",
                      "borderRadius": 15,
                      "flexDirection": "row",
                      "justifyContent": "center",
                      "marginTop": 5,
                      "padding": 15,
                      "width": "100%",
                    }
                  }
                >
                  <Text
                    style={
                      Object {
                        "color": "#fff",
                        "fontSize": 14,
                        "fontWeight": "bold",
                        "textTransform": "uppercase",
                      }
                    }
                  >
                    Login
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </View>
      </RCTScrollView>
    </View>
  </View>
</RCTSafeAreaView>
`;

exports[`Register screen renders correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Array [
      Object {
        "flex": 1,
        "marginTop": 20,
      },
      Object {
        "flex": 1,
        "marginTop": 0,
      },
    ]
  }
>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      accessibilityIgnoresInvertColors={true}
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <Image
        blurRadius={0.9}
        source={1}
        style={
          Array [
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
            Object {
              "height": undefined,
              "width": undefined,
            },
            undefined,
          ]
        }
      />
      <RCTScrollView>
        <View>
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#f5f5f5",
                "borderColor": "#B5A197",
                "borderWidth": 10,
                "elevation": 20,
                "height": 700,
                "justifyContent": "center",
                "margin": 20,
                "padding": 10,
                "shadowColor": "#000",
              }
            }
          >
            <View>
              <Text
                style={
                  Object {
                    "fontFamily": "Roboto",
                    "fontSize": 25,
                  }
                }
              >
                Register
              </Text>
            </View>
            <View
              style={
                Object {
                  "marginBottom": 30,
                  "marginTop": 50,
                }
              }
            >
              <View
                style={
                  Object {
                    "backgroundColor": "#e0e0eb",
                    "borderRadius": 25,
                    "flexDirection": "row",
                    "marginVertical": 10,
                    "padding": 10,
                    "width": "100%",
                  }
                }
              >
                <Text />
                <TextInput
                  allowFontScaling={true}
                  autoCapitalize="none"
                  autoCorrect={false}
                  keyboardType="email-address"
                  onBlur={[Function]}
                  onChangeText={[Function]}
                  placeholder="Email Address"
                  rejectResponderTermination={true}
                  style={
                    Object {
                      "flex": 1,
                      "fontFamily": "Roboto",
                      "fontSize": 20,
                      "marginLeft": 10,
                    }
                  }
                  underlineColorAndroid="transparent"
                  value=""
                />
              </View>
              <View
                style={
                  Object {
                    "backgroundColor": "#e0e0eb",
                    "borderRadius": 25,
                    "flexDirection": "row",
                    "marginVertical": 10,
                    "padding": 10,
                    "width": "100%",
                  }
                }
              >
                <Text />
                <TextInput
                  allowFontScaling={true}
                  autoCapitalize="none"
                  autoCorrect={false}
                  keyboardType="default"
                  onBlur={[Function]}
                  onChangeText={[Function]}
                  placeholder="Username"
                  rejectResponderTermination={true}
                  style={
                    Object {
                      "flex": 1,
                      "fontFamily": "Roboto",
                      "fontSize": 20,
                      "marginLeft": 10,
                    }
                  }
                  underlineColorAndroid="transparent"
                  value=""
                />
              </View>
              <View
                style={
                  Object {
                    "backgroundColor": "#e0e0eb",
                    "borderRadius": 25,
                    "flexDirection": "row",
                    "marginVertical": 10,
                    "padding": 10,
                    "width": "100%",
                  }
                }
              >
                <Text />
                <TextInput
                  allowFontScaling={true}
                  autoCapitalize="words"
                  autoCorrect={false}
                  keyboardType="default"
                  onBlur={[Function]}
                  onChangeText={[Function]}
                  placeholder="Full Name"
                  rejectResponderTermination={true}
                  style={
                    Object {
                      "flex": 1,
                      "fontFamily": "Roboto",
                      "fontSize": 20,
                      "marginLeft": 10,
                    }
                  }
                  underlineColorAndroid="transparent"
                  value=""
                />
              </View>
              <View
                style={
                  Object {
                    "backgroundColor": "#e0e0eb",
                    "borderRadius": 25,
                    "flexDirection": "row",
                    "marginVertical": 10,
                    "padding": 10,
                    "width": "100%",
                  }
                }
              >
                <Text />
                <TextInput
                  allowFontScaling={true}
                  autoCapitalize="none"
                  autoCorrect={false}
                  keyboardType="default"
                  onBlur={[Function]}
                  onChangeText={[Function]}
                  placeholder="Password"
                  rejectResponderTermination={true}
                  secureTextEntry={true}
                  style={
                    Object {
                      "flex": 1,
                      "fontFamily": "Roboto",
                      "fontSize": 20,
                      "marginLeft": 10,
                    }
                  }
                  underlineColorAndroid="transparent"
                  value=""
                />
              </View>
            </View>
            <View
              style={
                Object {
                  "marginBottom": 10,
                }
              }
            >
              <View
                accessible={true}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "opacity": 1,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "#BBB6A5",
                      "borderRadius": 15,
                      "flexDirection": "row",
                      "justifyContent": "center",
                      "marginTop": 5,
                      "padding": 15,
                      "width": "100%",
                    }
                  }
                >
                  <Text
                    style={
                      Object {
                        "color": "#fff",
                        "fontSize": 14,
                        "fontWeight": "bold",
                        "textTransform": "uppercase",
                      }
                    }
                  >
                    Register
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </View>
      </RCTScrollView>
    </View>
  </View>
</RCTSafeAreaView>
`;

exports[`Welcome screen renders correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Array [
      Object {
        "flex": 1,
        "marginTop": 20,
      },
      undefined,
    ]
  }
>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      accessibilityIgnoresInvertColors={true}
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <Image
        blurRadius={0.9}
        source={1}
        style={
          Array [
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
            Object {
              "height": undefined,
              "width": undefined,
            },
            undefined,
          ]
        }
      />
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#f5f5f5",
            "borderColor": "#B5A197",
            "borderWidth": 10,
            "elevation": 20,
            "flex": 0.7,
            "justifyContent": "center",
            "margin": 20,
            "marginTop": "40%",
            "shadowColor": "#000",
          }
        }
      >
        <Text />
        <View
          style={
            Object {
              "fontFamily": "Roboto",
            }
          }
        >
          <View>
            <Text
              style={
                Object {
                  "fontFamily": "Roboto",
                  "fontSize": 25,
                }
              }
            >
              Welcome to memorybook!
            </Text>
          </View>
          <Text
            style={
              Array [
                Object {
                  "fontFamily": "Roboto",
                  "fontSize": 20,
                },
                Object {
                  "alignItems": "center",
                  "fontStyle": "italic",
                  "justifyContent": "center",
                },
              ]
            }
          >
            "One Solution to storing photos"
          </Text>
        </View>
        <View
          style={
            Object {
              "margin": 10,
            }
          }
        >
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#BBB6A5",
                  "borderRadius": 15,
                  "flexDirection": "row",
                  "justifyContent": "center",
                  "marginTop": 5,
                  "padding": 15,
                  "width": "100%",
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "#fff",
                    "fontSize": 14,
                    "fontWeight": "bold",
                    "textTransform": "uppercase",
                  }
                }
              >
                Login
              </Text>
            </View>
          </View>
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "#BBB6A5",
                    "borderRadius": 15,
                    "flexDirection": "row",
                    "justifyContent": "center",
                    "marginTop": 5,
                    "padding": 15,
                    "width": "100%",
                  },
                  Object {
                    "backgroundColor": "#8C8C9C",
                  },
                ]
              }
            >
              <Text
                style={
                  Object {
                    "color": "#fff",
                    "fontSize": 14,
                    "fontWeight": "bold",
                    "textTransform": "uppercase",
                  }
                }
              >
                Register
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</RCTSafeAreaView>
`;
